File: 프로그래머스/unrated/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    num_len = len(p)
    base = int(p)
    return sum(num <= base for num in generate_num(t,num_len))

def generate_num(t, num_len):
    num = t[:num_len]
    yield int(num)
    for digit in t[num_len:]:
        num = num[1:]+digit
        yield int(num)

File: 프로그래머스/unrated/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.js
function solution(t, p) {
    const numLen = p.length
    let num = Array.from({length: numLen-1}, (v,i)=>t[i]).join('')
    let res = 0
    for (let idx = numLen-1; idx < t.length; idx++) {
        num = num + t[idx]
        if (num <= p) res++
        num = num.slice(1)
    }
    return res
}